src/system/CMakeLists.txt

```diff
@@ -18,20 +18,33 @@ if(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
 endif(WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD OR WITH_SYSTEMD_JOURNAL)
 
 set(dlt_system_SRCS dlt-system.c dlt-system-options.c dlt-system-process-handling.c
-	dlt-system-filetransfer.c dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
+   dlt-system-logfile.c dlt-system-processes.c dlt-system-shell.c
 	dlt-system-syslog.c dlt-system-watchdog.c dlt-system-journal.c)
+
+ if(WITH_DLT_FILETRANSFER)
+  set(dlt_system_SRCS ${dlt_system_SRCS} dlt-system-filetransfer.c)
+  add_definitions(-DDLT_FILETRANSFER_ENABLE)
+  message( STATUS "Building dlt-system WITH filetransfer support")
```

Comment:
We should put the STATUS on the STATUS section in top CMakeLists.txt at line 239 (after WITH_DLT_SYSTEM), so we can remove this one and line 29 as well.